Themis/API/ThemisAPI.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  ThemisAPI.swift
+//  Themis
+//
+//  Created by Andreas Cselovszky on 25.01.23.
+//
+
+import Foundation
+import CodeEditor
+
+enum ThemisAPI {
+    static let restController = RESTController(baseURL: URL(string: themisServer ?? "localhost")!)
+    
+    private static func buildAuthenticatedRequest(_ request: Request) throws -> Request {
+        var request = request
+        if Authentication.shared.isBearerTokenAuthNeeded() {
+            guard let token = Authentication.shared.token else {
+                throw Authentication.AuthenticationError.tokenNotFound
+            }
+            request.headers["Authorization"] = token
+        } else {
+            if let cookie = HTTPCookieStorage.shared.cookies(for: RESTController.shared.baseURL)?.first {
+                request.headers["Authorization"] = cookie.value
+            }
+        }
+        return request
+    }
+    
+    static func sendRequest<T: Decodable>(_ type: T.Type, request: Request) async throws -> T {
+        try await restController.sendRequest(
+            buildAuthenticatedRequest(request)
+        )
+    }
+
+    static func sendRequest<T>(_ type: T.Type, request: Request, decode: (Data) throws -> T) async throws -> T {
+        try await restController.sendRequest(
+            buildAuthenticatedRequest(request),
+            decode: decode)
+    }
+
+    static func sendRequest(request: Request) async throws {
+        try await restController.sendRequest(
+            buildAuthenticatedRequest(request)
+        )
+    }
+}
+
+struct NotifyRequest: Codable {
+    let exercise_id: Int
+    let participation_id: Int
+    let server: String
+}
+
+struct FeedbackSuggestionRequest: Codable {
+    let server: String
+    let exercise_id: Int
+    let participation_id: Int
+}
+
+extension ThemisAPI {
+    /// notifies Themis-ML about new feedback that should be pulled
+    static func notifyAboutNewFeedback(exerciseId: Int, participationId: Int) async throws {
+        let request = Request(
+            method: .post,
+            path: "/feedback_suggestion/notify",
+            body: NotifyRequest(
+                exercise_id: exerciseId,
+                participation_id: participationId,
+                server: RESTController.shared.baseURL.absoluteString
+            )
+        )
+        try await restController.sendRequest(request)
+    }
+    
```

Comment:
Just a question: Should there be a third resource, where our app sends the given feedback to Themis-ML? Otherwise the Themis-ML database does not fill with feedbacks, correct? Or is this not in the scope of the story?

Reply:
I don't really understand your question. Themis notifies Themis-ML about new feedbacks and Themis-ML will pull the new feedbacks from Artemis. Maybe this answers it.

Reply:
I'm pretty sure that this [endpoint](https://github.com/ls1intum/Themis-ML/blob/feature/get-feedback-suggestions/feedback-suggestion/src/app/endpoints/FeedbackSuggestionRequest.py#:~:text=%40router.post(%22/feedback_suggestion/notify%22)) not only notifies Themis-ML if there is a new feedback, but also saves it to the database if the feedback is not in the database yet.

Reply:
I probably meant what @crustypluto19 explained. I thought that there is no endpoint where we save the feedback we gave and submitted but the first one actually does that. Thanks!